internal/scheduler/api.go

```diff
@@ -3,6 +3,7 @@ package scheduler
 import (
 	"context"
```

Comment:
Please ignore changes in this file- this is not yet used by the scheduler and I plan to update it in a subsequent PR
For now, I've done enough to make compilation work but no more.